data/sea/52-psv.h

```diff
@@ -142,10 +148,9 @@ static ierror_msg_t psv_read_buffer (psv_state_t *s)
         const char  *value_end  = time_ptr - 1;
         const size_t value_size = value_end - value_ptr;
 
-        const bool new_entity = entity_cur_size != entity_size
-                             || memcmp (entity_cur, entity_ptr, entity_size) != 0;
+        const int new_entity = memcmp (entity_cur, entity_ptr, entity_size);
```

Comment:
I'm not sure about this. I think you need to find the minimum size, then if it's 0 subtract the two sizes.

suppose the old entity is ``ABC`` and the new one is ``AB``. I think this would compare with the pipe, so ``memcmp("ABC", "AB|", 3)`` which would say ``AB|`` is larger than ``ABC``, pipe being ascii 124 apparently

does that make sense?